tests/unit/test_callable_generator.py

```diff
@@ -40,8 +41,7 @@ def test_mapper_from_graph(self):
         t = CallableGeneratorTest.Temp()
 
         # Head node
-        node = Node(None, None)
-
+        node = TransformationProxy(Node(None, None))
```

Comment:
Since `node` is now a `Proxy`, in the line 53 of this same file, shouldn't we have something like this (even though it work now)? We want to pass the node instead of the proxy:

```
# Generate and execute the mapper
generator = CallableGenerator(node.proxied_node)
```

Reply:
So the `CallableGenerator` accesses attributes and methods of the node(s), such as `graph_prune` or `operation`. Since (transformation) proxies return all attributes and methods of their proxied node, the tests now works regardless of what object we pass. 

Being said that, I think that as it is now it double checks both the fact that the proxy returns the correct attributes and methods of its node and that the node behaves correctly. Nevertheless, this makes it less clear that what the `CallableGenerator` class handles are nodes, not proxies.

Reply:
In the end, some more clarity can play at our advantage. I will got with the call to the proxied node